conan/tools/cmake/utils.py

```diff
@@ -4,14 +4,22 @@ def is_multi_configuration(generator):
     return "Visual" in generator or "Xcode" in generator or "Multi-Config" in generator
 
 
-def get_file_name(conanfile, find_module_mode=False):
+def get_file_name(conanfile, forced_module_mode=None):
     """Get the name of the file for the find_package(XXX)"""
     # This is used by the CMakeToolchain to adjust the XXX_DIR variables and the CMakeDeps. Both
     # to know the file name that will have the XXX-config.cmake files.
-    if find_module_mode:
+    if forced_module_mode or get_find_mode(conanfile) == "module":
         ret = conanfile.cpp_info.get_property("cmake_module_file_name")
         if ret:
             return ret
     ret = conanfile.cpp_info.get_property("cmake_file_name")
     return ret or conanfile.ref.name
 
+
+def get_find_mode(conanfile):
+    """
+    :param conanfile: conanfile of the requirement
+    :return: "none" or "config" or "module" or "both"
+    """
+    tmp = conanfile.cpp_info.get_property("cmake_find_mode") or "both"
```

Comment:
Documentation [here](https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html) says, that `cmake_find_mode` defaulted to "config". So should it be `or "config"?

Reply:
Mhh not in this case, but yes, this is a bit messy so I will return None when not set.

Reply:
Thanks for including comment in the "convertation to `{{ _DEPENDENCY }} _FIND_MODE`": now I understand this!